### DISCUSSION TITLE
having some probs using LDAP (mailserver + docker.io/bitnami/openldap:2.6)... would need some conf examples.

### DISCUSSION THREAD

**Discussion by fab-kaz** [docker-mailserver/docker-mailserver#3089] [Q&A]
Hi folks,

For a few days, we've been trying to setup ldap with mailserver with no luck.

The ldap server works ok (with ltbproject/self-service-password as a web client). The web client works ok too, we can change passwords in the ldap directory.

Our ldap access rules looks like this:
```
dn: olcDatabase={2}mdb,cn=config
changeType: modify
replace: olcAccess
olcAccess: {0}to attrs=userPassword,shadowLastChange by self write by anonymous auth by dn="cn=admin,dc=my_domain,dc=fr" write by * none
olcAccess: {1}to dn.base="" by * read
olcAccess: {2}to dn.base="ou=users,dc=my_domain,dc=fr" by users read
olcAccess: {3}to * by self write by dn="cn=admin,dc=my_domain,dc=fr" write by * read
```
Here is one example of a ldapsearch:
```
# ldapsearch -x -b "dc=my_domain,dc=fr" "mail=john@my_domain.fr"

# john@my_domain.fr, users, my_domain.fr
dn: cn=john@my_domain.fr,ou=users,dc=my_domain,dc=fr
objectClass: inetOrgPerson
sn: john@my_domain.fr
mail: john@my_domain.fr
cn: john@my_domain.fr
```
Well, all of this to say ldap server and web client are working ok.

Now with mailserver/dovecot conf. Here is /etc/dovecot/dovecot-ldap.conf.ext
```
base = ou=users,dc=my_domain,dc=fr
default_pass_scheme = CRYPT
dn = cn=admin,dc=my_domain,dc=fr
dnpass = admin_pass_very_complicated
uris = ldap://ldap
tls = no
ldap_version = 3
pass_attrs = uniqueIdentifier=mail,userPassword=password
pass_filter = (&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))
user_attrs = mailHomeDirectory=home,mailUidNumber=uid,mailGidNumber=gid,mailStorageDirectory=mail
user_filter = (&(objectClass=PostfixBookMailAccount)(uniqueIdentifier=%n))
auth_bind = yes
```
We don't use yet the LDAP variables in the docker-compose because it seems some of them are missing (ie for the variable auth_bind).

When we restart postfix/dovecot with this conf and try to access mail account with a roundcube, we get:
```
ldapServ | conn=1320 op=6 SRCH base="ou=users,dc=my_domain,dc=fr" scope=2 deref=0 filter="(&(?objectClass=PostfixBookMailAccount)(uniqueIdentifier=john))"
```

I wonder why the unique identifier is "john". It should be john@my_domain.fr ? 

Thank you for all people that can share postfix/dovecot conf ;)

regards,

f.

**Comment by georglauterbach** [comment]
There are currently no LDAP-Code maintainers, hence the code is really not the best. We are actively looking though 😉

If you see a misconfiguration from our side, please open a bug report :)

**Comment by polarathene** [comment]
As mentioned, no current maintainer has experience with LDAP to be of much help.

- [This is our only test coverage for LDAP support](https://github.com/docker-mailserver/docker-mailserver/blob/16f46c1e1465212b7bd648c9690896cdf6c9a2e8/test/tests/serial/mail_with_ldap.bats)
- It manages the [LDAP setup with this Dockerfile and config files](https://github.com/docker-mailserver/docker-mailserver/tree/a0791ef45744f9e0ff15be7cbdcad455d249494f/test/docker-openldap)

Perhaps you can observe some difference from those files, or it has the same problem? The Dockerfile hasn't been updated in quite some time so the LDAP image used is rather outdated as well.

---

Beyond that, there's a few pages in our docs with LDAP config, and you can search past issues to find a bunch of other LDAP users sharing config/advice that may be helpful.

Please do share here if you're successful troubleshooting, it would benefit other LDAP users :)
